### PR TITLE
Felix/track current trajectory

### PR DESCRIPTION
**Pull request by felixerdy**
Adding mapbox-gl to tracking page to see current trajectory. Using TrajectoryService and needs feature as described in #17. 

Issue ref: #7 

**Note**: Don't forget to set your mapbox access token in `src/environments/environment(.prod).ts`

### CODE REVIEW

File: src/app/shared-services/db/sqlite.service.ts [hunk 1]
```diff
@@ -126,9 +126,16 @@ export class SqliteService {
       message,
     } = await this.db.run({
       statement: 'INSERT OR REPLACE INTO points VALUES (?,?,?,?,?)',
-      values: [trajectoryId, time.toISOString(), ...p.latLng, p.accuracy],
+      // IDK why but with iOS simulator it was not possible to run this without .toFixed()
+      values: [
+        trajectoryId,
+        time.toISOString(),
+        p.latLng[1].toFixed(8),
+        p.latLng[0].toFixed(8),
```
Comment (noerw) [inline src/app/shared-services/db/sqlite.service.ts:133]: coords are in reverse order right? points table is sorted lat first, lon second

Comment (felixerdy) [reply src/app/shared-services/db/sqlite.service.ts:133]: Ah I missed that mapbox is using the reversed way. I will resolve that

**Comment by zven** [comment]
When I was trying to run the current development state on iOS I ran into this issue:
Screenshot 2020-11-22 at 14 31 35

@felixerdy Is that the sql problem you were referring to in one of the commit messages?

**Comment by zven** [comment]
> When I was trying to run the current development state on iOS I ran into this issue:
> Screenshot 2020-11-22 at 14 31 35
> 
> @felixerdy Is that the sql problem you were referring to in one of the commit messages?

I just tried it myself and your changes do seem to fix the problem.

**Comment by noerw** [comment]
What's the idea behind adding a map view to the tracking page? Is this intended as a preview?
I mean, we already have the trajectory view with map & inferences etc at `/trajectory/track/user`.

Current UX is kinda weird, you enable tracking, go back and open an "example" trajectory, and select "Your Trajectory"..
Maybe add a button "analyze my trajectory" below the tracking-toggle on the tracking page. It links to `/trajectory/track/user`. 

Then we have 2 options regarding the trajectory listing:
1. "your trajectory" is not listed in the "view example trajectories" list anymore.
2. "view example trajectories" button is renamed to something more generic, like "analyze trajectories"

**Comment by noerw** [comment]
Meanwhile I'll cherrypick your iOS fix into develop, please rebase/merge

**Comment by felixerdy** [comment]
> What's the idea behind adding a map view to the tracking page? Is this intended as a preview?
> I mean, we already have the trajectory view with map & inferences etc at `/trajectory/track/user`.

I thought that was the idea behind issue #7 

> 
> Current UX is kinda weird, you enable tracking, go back and open an "example" trajectory, and select "Your Trajectory"..
> Maybe add a button "analyze my trajectory" below the tracking-toggle on the tracking page. It links to `/trajectory/track/user`.
> 
> Then we have 2 options regarding the trajectory listing:
> 
> 1. "your trajectory" is not listed in the "view example trajectories" list anymore.
> 2. "view example trajectories" button is renamed to something more generic, like "analyze trajectories"

I agree that UX is currently weird. Also: do we need a separate page to start/stop recording? Maybe we could discuss that in a new issue and mark this PR as a draft (?)

**Comment by zven** [comment]
Reading this discussion I realized that the user trajectory is actually _not_ showing up in the list of available trajectories on iOS. 🤔

**Comment by zven** [comment]
> Reading this discussion I realized that the user trajectory is actually _not_ showing up in the list of available trajectories on iOS. 🤔

Fixed on development branch: https://github.com/sitcomlab/simport-learning-app/commit/256e9b4e0705b216005bbb7dc425e7b607ade97e

**Comment by zven** [comment]
We can delete this branch / pull request, right?

**Pull request closed without merging**